pkg/extensions/cluster.go

```diff
@@ -27,13 +27,84 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/runtime/serializer"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
-var gardenscheme *runtime.Scheme
+var gardenScheme *runtime.Scheme
 
 func init() {
-	gardenscheme = runtime.NewScheme()
-	gardencoreinstall.Install(gardenscheme)
+	gardenScheme = runtime.NewScheme()
+	gardencoreinstall.Install(gardenScheme)
+}
+
+// NewGardenDecoder returns a new Garden API decoder.
+func NewGardenDecoder() (runtime.Decoder, error) {
+	return serializer.NewCodecFactory(gardenScheme).UniversalDecoder(), nil
+}
+
+// SyncClusterResourceToSeed creates or updates the `extensions.gardener.cloud/v1alpha1.Cluster` resource in the seed
+// cluster by adding the shoot, seed, and cloudprofile specification.
+func SyncClusterResourceToSeed(
+	ctx context.Context,
+	client client.Client,
+	clusterName string,
+	shoot *gardencorev1beta1.Shoot,
+	cloudProfile *gardencorev1beta1.CloudProfile,
+	seed *gardencorev1beta1.Seed,
+) error {
+	if shoot.Spec.SeedName == nil {
+		return nil
+	}
+
+	var (
+		cluster = &extensionsv1alpha1.Cluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: clusterName,
+			},
+		}
+
+		cloudProfileObj *gardencorev1beta1.CloudProfile
+		seedObj         *gardencorev1beta1.Seed
+		shootObj        *gardencorev1beta1.Shoot
+	)
+
+	if cloudProfile != nil {
+		cloudProfileObj = cloudProfile.DeepCopy()
+		cloudProfileObj.TypeMeta = metav1.TypeMeta{
+			APIVersion: gardencorev1beta1.SchemeGroupVersion.String(),
+			Kind:       "CloudProfile",
+		}
```

Comment:
What do you think about purging the `ManagedFields` here? (also for the seed and shoot).
```suggestion
		cloudProfileObj = cloudProfile.DeepCopy()
		cloudProfileObj.TypeMeta = metav1.TypeMeta{
			APIVersion: gardencorev1beta1.SchemeGroupVersion.String(),
			Kind:       "CloudProfile",
		}
		cloudProfileObj.ObjectMeta.ManagedFields = nil
```